### PR TITLE
Added Ancient Guardians species as an alternative to Sentries guardin…

### PR DESCRIPTION
**Pull request by Deepsloth**
…g specials.

### CODE REVIEW

**Comment by Deepsloth** [comment]
Discussion: http://www.freeorion.org/forum/viewtopic.php?f=15&t=10162

Post v0.4.6

File: default/stringtables/en.txt [hunk 1]
```diff
@@ -5129,6 +5129,11 @@ On %planet%: Ancient Ruins were excavated, but only tantalizing hints of ancient
 EFFECT_ANCIENT_RUINS_EMPTY_LABEL
 Ancient Ruins Empty Excavation
 
+EFFECT_ANCIENT_GUARDIANS_CAPTURED
+On %planet%: The [[species SP_ANCIENT_GUARDIANS]] self-destructed when the planet was captured!
+EFFECT_ANCIENT_GUARDIANS_CAPTURED_LABEL
+Ancient Guardians seld destruct
```
Comment (MatGB) [inline default/stringtables/en.txt]: self

File: default/stringtables/en.txt [hunk 2]
```diff
@@ -6197,6 +6202,18 @@ SP_EXOBOT_DESC
 '''While not as efficient as fully self-directed citizens, Exobots can colonize otherwise inhospitable planets, and build more Exobots.
 '''
 
+SP_ANCIENT_GUARDIANS
+Ancient Guardians
+SP_ANCIENT_GUARDIANS_GAMEPLAY_DESC
+'''Guardian Robots constructed by the Precursor civilizations aeons ago
```
Comment (MatGB) [inline default/stringtables/en.txt]: Better to say "a Precursor", we've got several in game already, no need to say which one.

File: default/stringtables/en.txt [hunk 3]
```diff
@@ -6197,6 +6202,18 @@ SP_EXOBOT_DESC
...
+[[encyclopedia ROBOTIC_SPECIES_TITLE]]
+Will self-destruct when conquered
+'''
+
+SP_ANCIENT_GUARDIANS_DESC
+'''Constructed by precusor technology these ancient machines can adapt to and fight in any kind of environment. With unwavering loyalty and duteousness they have protected the planet from any invaders, never wasting time to repair and improve the fortifications. Upon being captured every Guardian Robot will automatically self-destruct.
```
Comment (MatGB) [inline default/stringtables/en.txt]: I dislike "duteousness", I had to look up whether it was an actual word and it's very rarely used (so much so Google won't even give me an ngram of it, insisting to give me results for duteous instead). "Commitment" might be better, not sure tho.

**Comment by MatGB** [comment]
These do look very fun. Not sure about troops as a multiplier of target population, that prevents partial destruction from bombardment which ought to be a valid tactic, multiplier of current pop?

(it is of course another special giving Egassem a potential early advantage...)

**Comment by Deepsloth** [comment]
Troops are not dependent on target population. "Target.Population" is the current population of the target. The other would be "Target.TargetPopulation". I love FOCS.

**Comment by MatGB** [comment]
Good point, yeah, see, still not grokking FOCS properly.

File: default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    description = "SP_ANCIENT_GUARDIANS_DESC"
+    gameplay_description = "SP_ANCIENT_GUARDIANS_GAMEPLAY_DESC"
+
+    tags = [ "ROBOTIC" "AI_TAG_NO_INDUSTRY" "AI_TAG_NO_RESEARCH" "AI_TAG_AVERAGE_SUPPLY"]
+
+    foci = [
+        [[HAS_INDUSTRY_FOCUS]]
+        [[HAS_RESEARCH_FOCUS]]
+        [[HAS_GROWTH_FOCUS]]
+        [[HAS_MINING_FOCUS]]
```
Comment (MatGB) [inline default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt:12]: Mining is gone, I was wandering why I hadn't found any until I checked my logs.

File: default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+        [[AVERAGE_DEFENSE_TROOPS]]
+
+        // not for description
+
+        [[AVERAGE_PLANETARY_SHIELDS]]
+        [[AVERAGE_PLANETARY_DEFENSE]]
+
+        // huge bonus to planetary shields
+        EffectsGroup
+            description = "ANCIENT_PLANETARY_SHIELD"
```
Comment (MatGB) [inline default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt:32]: as of ec2f1cc this needs a _DESC suffix, @adrianbroher is on a bug hunt

File: default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+            scope = And [
+                Source
+                Planet
+            ]
+            effects = SetMaxShield value = Value + 500
+
+        // huge bonus to ground troops
+        EffectsGroup
+            description = "ANCIENT_DEFENSE_TROOPS"
```
Comment (MatGB) [inline default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt:41]: same for this

File: default/stringtables/en.txt [hunk 7]
```diff
@@ -11374,6 +11391,9 @@ GREAT_DEFENSE_TROOPS
 ULTIMATE_DEFENSE_TROOPS
 +++	Ultimate Defensive Ground [[encyclopedia TROOP_TITLE]]: 300%
 
+ANCIENT_DEFENSE_TROOPS
++++	Ancient Defensive Ground [[encyclopedia TROOP_TITLE]]: 10 per population
+
```
Comment (MatGB) [inline default/stringtables/en.txt:11396]: this and the below shield are where the merge conflicts were for me, I made a hash of sorting it out so better to be redone, both need the _desc suffix as above.

**Review by MatGB — CHANGES_REQUESTED** [review]
I like the idea of this and want to expand on it, but better to get this in then use it for more down the line, fairly sure it needs AI updates as well.

File: default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+            effects = SetMaxTroops value = Value + Target.Population * 10
+
+        // self-destruct when captured
+        EffectsGroup
+            scope = Source
+            activation = OwnedBy affiliation = AnyEmpire
+            effects = [
+                SetPopulation value = 0
+                GenerateSitRepMessage
+                    message = "EFFECT_ANCIENT_GUARDIANS_CAPTURED"
```
Comment (MatGB) [inline default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt:55]: Again, needs the label here.

**Comment by MatGB** [comment]
Think they need an effect that sets their planetary population to max at start of game, if they're meant to be ancient they should have max pop not be growing on game start, especially given the troop effects.

Beyond that, finally got them working with current backend and I really like. Moving forward, the AI team need to review and possibly update the AI.

After that's been done, I'd like to expand the concept: for balance reasons I'd like the chance of guard on most of the big specials to always apply, so if a sentinel isn't generated these guys always should be, and possibly with a bonus of some sort for the bigger monsters, I've been uncomfortable for ages that ancient ruins can sometimes give you a really powerful unlock early with no defence whatsoever, etc.  But that's an ongoing balance thing, these are cool and I like them as is instead of sentries.

**Comment by MatGB** [comment]
Also, Sitrep Priority needs to be added to the self destruct so it's just above the "dies out completely" message.

File: default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt [hunk 9]
```diff
@@ -0,0 +1,81 @@
+Species
+    name = "SP_ANCIENT_GUARDIANS"
+    description = "SP_ANCIENT_GUARDIANS_DESC"
+    gameplay_description = "SP_ANCIENT_GUARDIANS_GAMEPLAY_DESC"
+
+    tags = [ "ROBOTIC" "AI_TAG_NO_INDUSTRY" "AI_TAG_NO_RESEARCH" "AI_TAG_AVERAGE_SUPPLY"]
```
Comment (Morlic-fo) [inline default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt:6]: The AI_TAG_ prefix was removed in e39340464446e7bd3b268088ac4f0d03393a20e1. Just use "NO_INDUSTRY" etc.

File: default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt (hunk 4 above)
Comment (Morlic-fo) [reply default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt:12]: Does it make sense for a non-playable, non-conquerable species to have any foci?

File: default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt (hunk 9 above)
Comment (Morlic-fo) [reply default/scripting/species/SP_ANCIENT_GUARDIANS.focs.txt:6]: Also I'd like to have a new tag "DESTROYED_ON_CONQUEST" for AI purposes.

File: default/scripting/specials/specials.macros [hunk 10]
```diff
@@ -29,7 +29,16 @@ CHANCE_OF_GUARD_1
                     And [ Planet HasSpecial name = "HIGH_TECH_NATIVES_SPECIAL" ]
                 ]
             ]
-            effects = CreateShip designname = OneOf("SM_GUARD_0", "SM_GUARD_1")
+            effects = [
+                If condition = Random probability = 0.6
+                    effects = [
+                        CreateShip designname = OneOf("SM_GUARD_0", "SM_GUARD_1")
+                    ]
```
Comment (Morlic-fo) [inline default/scripting/specials/specials.macros:36]: Any reason to use brackets and 3 lines for a singe-line-statement?

File: default/stringtables/en.txt [hunk 11]
```diff
@@ -6197,6 +6202,18 @@ SP_EXOBOT_DESC
...
+SP_ANCIENT_GUARDIANS
+Ancient Guardians
+SP_ANCIENT_GUARDIANS_GAMEPLAY_DESC
+'''Guardian Robots constructed by a Precursor civilization aeons ago
+[[encyclopedia ROBOTIC_SPECIES_TITLE]]
+Will self-destruct when conquered
+'''
+
+SP_ANCIENT_GUARDIANS_DESC
+'''Constructed by precusor technology these ancient machines can adapt to and fight in any kind of environment. With unwavering loyalty and commitment they have protected the planet from any invaders, never wasting time to repair and improve the fortifications. Upon being captured every Guardian Robot will automatically self-destruct.
```
Comment (Morlic-fo) [inline default/stringtables/en.txt:6214]: Not a native speaker - shouldn't there be a comma after `Constructed by precusor technology`, `With unwavering loyalty and commitment`, and `Upon being captured`?

**Review by Morlic-fo — CHANGES_REQUESTED** [review]
This will need some AI adaptations but I wouldn't consider those to be critical enough to delay the merge.

@MatGB As before, please open another AI issue when you merge this.

**Comment by Morlic-fo** [comment]
> Moving forward, the AI team need to review and possibly update the AI.
> 
> After that's been done, I'd like to expand the concept: for balance reasons I'd like the chance of guard on most of the big specials to always apply

While this PR may be merged without the AI adaptations, the concept should not be expanded until the relevant AI changes have been made. A rare special that the AI acts weird on is fine. Having half the galaxy being populated by Ancient Guardians is a different matter...

File: default/stringtables/en.txt (hunk 11 above)
Comment (MatGB) [reply default/stringtables/en.txt:6214]: I am, but am terrible with punctuation, I checked with the editor-I-live-with and she says yes in all three, the latter two being optional for reading-as-text but required if to be read aloud. The first definitely as it's two clauses.

I wouldn't have put any of them in either, so good catch.

**Comment by MatGB** [comment]
> A rare special that the AI acts weird on is fine. Having half the galaxy being populated by Ancient Guardians is a different matter...

Fair enough: what I'm hoping for is that the existing chance-of-guard becomes chance of guard monster or chance of getting these, so that everything that gives a decent bonus gets something. But it's a longer term thing and is only a balance quibble not a major concern, it would also allow us to maintain having ancient ruins in No Monsters galaxies without me having conniptions over the balance issues there (ie the chance of a monster is determined by monster setting else you get these)

**Comment by Deepsloth** [comment]
I made a new branch with a fixed and tweaked version of the ancient guardians, but there is one requested change i haven't figured out:

"Also, Sitrep Priority needs to be added to the self destruct so it's just above the "dies out completely" message."

How do i implement that?

**Comment by dbenage-cx** [comment]
For now: https://github.com/freeorion/freeorion/blob/master/default/customizations/common_user_customizations.txt contains an ordered list of sitreps under `FUNCTIONAL_SITREP_PRIORITY_ORDER`.  These sitreps are displayed before those with no entry.

**Pull request closed without merging**